client/actions/index.js

```diff
@@ -1,74 +1,81 @@
-export function updateSearchTitle(title) {
+export function updateVerb(verb) {
   return {
-    type: 'UPDATE_SEARCH_TITLE',
+    type: 'UPDATE_VERB',
     payload: {
-      title,
+      verb,
     },
   };
 }
 
-export function updateSearchLimit(limit) {
+export function updatePath(path) {
   return {
-    type: 'UPDATE_SEARCH_LIMIT',
+    type: 'UPDATE_PATH',
     payload: {
-      limit,
+      path,
     },
   };
 }
 
-export function searchAction(searchFields) {
-  const { title, limit } = searchFields;
-  let params = `limit=${limit}`;
-  if (title.length) {
-    params += `&title=${title}`;
+export function updateParams(params) {
+  return {
+    type: 'UPDATE_PARAMS',
+    payload: {
+      params,
+    },
+  };
+}
+
+export function sendRequest(requestFields) {
+  const { verb, path, params } = requestFields;
+
+  const fetchOptions = {
+    method: verb,
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    },
+    credentials: 'include',
+  }
+
+  if (verb !== 'GET') {
+    fetchOptions['body'] = params
   }
```

Comment:
I'd like to find a cleaner way to handle this. `fetch` threw an error if I added `body` to a GET request

Reply:
Probably we just want to disable the body box or collapse it when they want to do a get request, right?

Reply:
We probably should do that. Otherwise its a bit weird to just not include it. This works fine with our API too because any filtering etc is done in a query param so it would go in the `path` input anyways. 

E.g.

```
GET /admin/products.json?collection_id=841564295
```